platform/logging/LoggingConfig.ts

```diff
@@ -0,0 +1,123 @@
+import * as schema from '../lib/schema';
+import { typeOfSchema } from '../types';
+import { LogLevel } from './LogLevel';
+import { BaseAppenderConfig, BaseAppenderConfigSchemaType } from './appenders/base/BaseAppenderConfig';
+import { ConsoleAppenderConfig } from './appenders/console/ConsoleAppenderConfig';
+import { FileAppenderConfig } from './appenders/file/FileAppenderConfig';
+import { RollingFileAppenderConfig } from './appenders/rolling-file/RollingFileAppenderConfig';
+
+const CONTEXT_SEPARATOR = '::';
+const ROOT_CONTEXT_NAME = 'root';
+
+const KNOWN_APPENDERS = new Map<string, new (schema:  BaseAppenderConfigSchemaType) => BaseAppenderConfig>([
+  ['console', ConsoleAppenderConfig],
+  ['file', FileAppenderConfig],
+  ['rolling-file', RollingFileAppenderConfig]
+]);
+
+const createLoggerSchema = () => {
+  return schema.object({
+    appenders: schema.mapOf(schema.string(), schema.oneOf([
+      ConsoleAppenderConfig.getSchema(),
+      FileAppenderConfig.getSchema(),
+      RollingFileAppenderConfig.getSchema()
+    ])),
+    loggers: schema.mapOf(schema.string(), schema.object({
+      level: schema.oneOf([
+        schema.literal('all'),
+        schema.literal('fatal'),
+        schema.literal('error'),
+        schema.literal('warn'),
+        schema.literal('info'),
+        schema.literal('debug'),
+        schema.literal('trace'),
+        schema.literal('off')
+      ], {
+        defaultValue: 'info'
+      }),
+      appenders: schema.arrayOf(schema.string(), { defaultValue: [] })
+    }), {
+      validate(value) {
+        const rootLogger = value.get(ROOT_CONTEXT_NAME);
+        if (!rootLogger) {
+          throw new Error(`"${ROOT_CONTEXT_NAME}" logger should be configured!`);
+        }
+
+        if (rootLogger.appenders.length === 0) {
+          throw new Error(`"${ROOT_CONTEXT_NAME}" logger should have at least one "appender" configured.`);
+        }
+      }
+    })
+  });
+};
+
+const loggingConfigType = typeOfSchema(createLoggerSchema);
+type LoggingConfigType = typeof loggingConfigType;
+
+export interface LoggerConfig {
+  appenders: string[];
+  level: LogLevel;
+}
+
+export class LoggingConfig {
+  static createSchema = createLoggerSchema;
+  readonly appenders: Map<string, BaseAppenderConfig> = new Map();
+  readonly loggers: Map<string, LoggerConfig> = new Map();
+
+  constructor(config: LoggingConfigType) {
+    this.fillAppendersConfig(config);
+    this.fillLoggersConfig(config);
```

Comment:
Ah, of course, didn't think about that. Right now the config service "owns" instantiating the `LoggingConfig`. Hm. Let's just keep the methods in there instead of moving them to a static helper for now. (Unless you have other ideas you want to explore, of course)